### PR TITLE
feat: simplify deeplink updates, use cloudflare worker

### PR DESCRIPTION
**Pull request by brainbicycle**
Resolves: https://artsyproduct.atlassian.net/browse/PHIRE-1143

Implements RFC here: https://www.notion.so/artsy/Simplify-universal-link-config-updates-fffcab0764a0804bbf32d730ee4ed134

Updates this repo to stop distributing a library to be consumed by clients.

Instead we will use the same cloudflare worker to distribute app-site-association and assetlinks for deeplinks on every url we need. Deployment will be similar to other repos, deploy to staging* (test worker) on merge to main, deploy to production using deploy prs.

**Implementation details:**
- use wrangler to deploy cloudflare workers 
- on merge to main deploy to a staging worker
- deploy prs to release to deploy to production same as other repos

### Follow-ups

- [ ] Once deployed and tested, update cloudflare config to serve this worker from all associated domains
- [ ] Remove the artsy-eigen-web-association dep from consuming repos

### CODE REVIEW

File: test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    worker = await unstable_dev("./worker.ts", {
+      experimental: { disableExperimentalWarning: true },
+    });
+  });
+
+  afterAll(async () => {
+    await worker.stop();
+  });
+
+  it("should return the apple-app-site-association JSON on the correct route", async () => {
```
Comment (damassi) [inline test.ts:42]: Nice @brainbicycle - I couldn't quite figure out how to test that other worker in MP 👌

**Review by damassi — APPROVED** [review]
This is such a solid / modern improvement over this whole complicated process. Dang! Very nice work 💯

File: test.ts (hunk 1 above)
Comment (brainbicycle) [reply test.ts:42]: Yeah i thought this was gonna be a pain but was pleasantly surprised! https://developers.cloudflare.com/workers/wrangler/api/

Comment (joeyAghion) [reply test.ts:42]: Oh I tried to get some basic test going in Metaphysics but was unsuccessful... possibly due to the worker's usage of the `cache` API.

**Review by joeyAghion — APPROVED** [review]

**Pull request merged**